src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -618,7 +616,13 @@ private void OnBrowseButtonClicked(object sender, EventArgs e)
             //const int BIF_RETURNONLYFSDIRS = 0x00000001;   // For finding a folder to start document searching.
             const int BIF_BROWSEINCLUDEURLS = 0x00000080; // Allow URLs to be displayed or entered.
 
-            var uiShell = (IVsUIShell2)_serviceProvider.GetService(typeof(SVsUIShell));
+            IVsUIShell2 uiShell = null;
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                uiShell = await AsyncServiceProvider.GlobalProvider.GetServiceAsync<IVsUIShell2>();
+            });
```

Comment:
As mentioned elsewhere, this doesn't make the code async, it just uses two threads instead of one.

Also, I'm not sure the async service provider needs the UI thread. I thought it doesn't, and therefore it's better to get services with it (or even the synchronous service provider) on background threads to avoid blocking the UI thread.

Reply:
If I understand correctly since code which is calling this one is itself is synchronous we better to keep old GetService instead of GetServiceAsync, is it correct? Otherwise we're just wrapping async code with JTF only increase overhead, but it still run as synchronous .
https://github.com/NuGet/NuGet.Client/blob/2c0c7c3f220833d2438c552df3cabeab24872297/src/NuGet.Clients/NuGet.Console/WpfConsole/WpfConsole.cs#L141-L142

Reply:
Ok. Reverted this change.